securedrop/journalist.py

```diff
@@ -242,9 +242,17 @@ def admin_reset_two_factor_hotp():
     otp_secret = request.form.get('otp_secret', None)
     if otp_secret:
         user = Journalist.query.get(uid)
-        user.set_hotp_secret(otp_secret)
-        db_session.commit()
-        return redirect(url_for('admin_new_user_two_factor', uid=uid))
+        try:
+            user.set_hotp_secret(otp_secret)
+            db_session.commit()
+            return redirect(url_for('admin_new_user_two_factor', uid=uid))
+        except TypeError as e:
+            if ("Non-hexadecimal digit found" not in str(e) and
+                "Odd-length string" not in str(e)):
+                raise
```

Comment:
the downside of showing a "unexpected error happened" is that it hides the exception completely. The error may not be useful to the user but it could be copy/pasted to a support ticket and be useful to the person doing the diagnostic. That being said, if that's really what you want I'll do it :-)